Regression from v2.2.6 to `version-2-2`/`devel` with importing `let u {.global.} = default(ref int)`

### Nim Version

Compiles:
```
Nim Compiler Version 2.2.4 [Linux: amd64]
Compiled at 2025-12-18
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: f7145dd26efeeeb6eeae6fff649db244d81b212d
active boot switches: -d:release
```

```
Nim Compiler Version 2.2.6 [Linux: amd64]
Compiled at 2025-12-18
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: ab00c56904e3126ad826bb520d243513a139436a
active boot switches: -d:release
```

Does  not compile:
```
Nim Compiler Version 2.2.7 [Linux: amd64]
Compiled at 2025-12-18
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: 244af30c587cb61a63219af916699529b51e4637
active boot switches: -d:release
```

```
Nim Compiler Version 2.3.1 [Linux: amd64]
Compiled at 2025-12-18
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: 9bb57a64ba63628219b67cb45cdc6f4fe38dc906
active boot switches: -d:release
```

Using either
```
gcc (Debian 15.2.0-11) 15.2.0
Copyright (C) 2025 Free Software Foundation, Inc.
```
or
```
Debian clang version 21.1.8 (1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-21/bin
```

### Description

`e.nim`:
```nim
proc v*() =
  let u {.global.} = default(ref int)
```

`j.nim`:
```nim
import ./e
v()
```

### Current Output

```text
gcc:
/tmp/e.nim(2, 7) Hint: 'u' is declared but not used [XDeclaredButNotUsed]
CC: system/exceptions.nim
CC: std/private/digitsutils.nim
CC: system/dollars.nim
CC: system.nim
CC: e.nim
CC: j.nim
Hint:  [Link]
/usr/bin/ld: /tmp/tmp.TeiMJJ6L1T/@mj.nim.c.o:(.bss+0x0): multiple definition of `u__e_u7'; /tmp/tmp.TeiMJJ6L1T/@me.nim.c.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
Error: execution of an external program failed: 'gcc   -o /tmp/j  /tmp/tmp.TeiMJJ6L1T/@psystem@sexceptions.nim.c.o /tmp/tmp.TeiMJJ6L1T/@pstd@sprivate@sdigitsutils.nim.c.o /tmp/tmp.TeiMJJ6L1T/@psystem@sdollars.nim.c.o /tmp/tmp.TeiMJJ6L1T/@psystem.nim.c.o /tmp/tmp.TeiMJJ6L1T/@me.nim.c.o /tmp/tmp.TeiMJJ6L1T/@mj.nim.c.o  -pthread -pthread   -ldl'

Or with clang:
/tmp/e.nim(2, 7) Hint: 'u' is declared but not used [XDeclaredButNotUsed]
CC: system/exceptions.nim
CC: std/private/digitsutils.nim
CC: system/dollars.nim
CC: system.nim
CC: e.nim
CC: j.nim
Hint:  [Link]
/usr/bin/ld: /tmp/tmp.ROolJXJ006/@mj.nim.c.o:(.bss+0x18): multiple definition of `u__e_u7'; /tmp/tmp.ROolJXJ006/@me.nim.c.o:(.bss+0x0): first defined here
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of an external program failed: 'clang   -o /tmp/j  /tmp/tmp.ROolJXJ006/@psystem@sexceptions.nim.c.o /tmp/tmp.ROolJXJ006/@pstd@sprivate@sdigitsutils.nim.c.o /tmp/tmp.ROolJXJ006/@psystem@sdollars.nim.c.o /tmp/tmp.ROolJXJ006/@psystem.nim.c.o /tmp/tmp.ROolJXJ006/@me.nim.c.o /tmp/tmp.ROolJXJ006/@mj.nim.c.o  -pthread -pthread   -ldl'
```

### Expected Output

```text
No C compilation/linking errors
```

### Known Workarounds

_No response_

### Additional Information

_No response_

--- a/compiler/injectdestructors.nim
+++ b/compiler/injectdestructors.nim
@@ -969,7 +969,7 @@ proc p(n: PNode; c: var Con; s: var Scope; mode: ProcessMode; tmpFlags = {sfSing
                       isInProc
 
               if isGlobalPragma:
-                c.graph.procGlobals.add n
+                c.graph.procGlobals.add newTree(nkFastAsgn, v, ri)
               else:
                 let value = moveOrCopy(v, ri, c, s, if v.kind == nkSym: {IsDecl} else: {})
                 result.add value

--- a/tests/global/mglobal3.nim
+++ b/tests/global/mglobal3.nim
@@ -1,0 +1,2 @@
+proc v*() =
+  let u {.global.} = default(ref int)

--- a/tests/global/tglobal3.nim
+++ b/tests/global/tglobal3.nim
@@ -62,3 +62,7 @@ proc m2()  =
   assert v == "123"
 
 m2()
+
+import mglobal3
+block:
+  v()

